backend/vehicle/views.py

```diff
@@ -105,8 +104,39 @@ def get(self, request):
             serialized_data = VehicleSerializer(vehicles, many=True)
 
             return Response(
-                {"vehicles": serialized_data.data,
-                    "more_data": has_more_data(request)}
+                {"vehicles": serialized_data.data, "more_data": has_more_data(request)}
             )
 
         return Response(VehicleSerializer(Vehicle.objects.all()[:10], many=True).data)
+
+
+class SaveUnitApiView(APIView):
+    """
+    An endpoint to handle saving a vehicle to a bidder's saved vehicle list
+    """
+
+    def get(self, request):
+        # Replace this line to obtain user id once authentication has been implemented
+        bidder_id = request.data.get("bidder_id")
+        try:
+            bidder = Bidder.objects.get(id=bidder_id)
+            vehicles = bidder.saved_list.all()
+            serialized_data = VehicleSerializer(vehicles, many=True)
+            return Response(serialized_data.data)
+        except Exception as e:
+            return Response("Bidder not found")
+
+    def post(self, request):
+        vehicle_id = request.data.get("vehicle_id")
+
+        # Replace this later to fetch authentication details from headers instead of body
+        bidder_id = request.data.get("bidder_id")
+
+        try:
+            vehicle = Vehicle.objects.get(id=vehicle_id)
+            bidder = Bidder.objects.get(id=bidder_id)
+
+            bidder.saved_list.add(vehicle)
+            return Response("Unit added successfully")
+        except Exception as e:
+            return Response("Bidder or vehicle not found")
```

Comment:
This code would make more sense in the `user` app since this is functionality for bidders